### PR TITLE
FateId to use UUID instead of long

### PR DESCRIPTION
**Pull request by kevinrr888**
closes https://github.com/apache/accumulo/issues/4277
Changes:
- FateId now uses a 128bit UUID instead of a (64bit) long
	- FateIds created by ZooStore and AccumuloStore now use a v4 UUID (random UUID)
	- FateIds created by FateIdGenerator now use a v3 UUID (name based) so the same FateKey gives the same UUID
- Necessary updates to classes and tests which used the long id

Note:
- Unfortunately, there is no simple way to include the creation time in the UUID of a FateId since FateIdGenerator requires that the same FateKey generates the same FateId. So, at least for now, FateId's UUID does not include a timestamp.

### CODE REVIEW

**Review by keith-turner — COMMENTED** [review]
Still looking at this

**Review by keith-turner — APPROVED** [review]
These changes look good, could not really review all the variable name changes which is why I asked if they were done w/ an ide.  Would be good to add the unit test for illegal fate ids.

**Comment by keith-turner** [comment]
I am going to open a follow on PR w/ the unit test changes I suggested

File: server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java [hunk 1]
```diff
@@ -584,7 +585,7 @@ protected TExternalCompactionJob createThriftJob(String externalCompactionId,
     }).collect(toList());
 
     FateInstanceType type = FateInstanceType.fromTableId(metaJob.getTabletMetadata().getTableId());
-    FateId fateId = FateId.from(type, 0);
+    FateId fateId = FateId.from(type, UUID.randomUUID());
```
[the remark this answers is not in the document]
Comment (keith-turner) [reply server/manager/src/main/java/org/apache/accumulo/manager/compaction/coordinator/CompactionCoordinator.java:588]: opened #4392

File: core/src/main/java/org/apache/accumulo/core/fate/FateId.java [hunk 2]
```diff
@@ -18,28 +18,29 @@
...
- * of the form "FATE:[FateInstanceType]:[hex long tid]" (without the brackets).
+ * Consists of its {@link FateInstanceType} and its transaction {@link UUID}. The canonical string
+ * is of the form "FATE:[FateInstanceType]:[UUID]" (without the brackets).
  */
 public class FateId extends AbstractId<FateId> {
 
   private static final long serialVersionUID = 1L;
   private static final String PREFIX = "FATE:";
-  private static final Pattern HEX_PATTERN = Pattern.compile("^[0-9a-fA-F]+$");
+  private static final String UUID_REGEX = "[0-9a-fA-F]{8}-([0-9a-fA-F]{4}-){3}[0-9a-fA-F]{12}";
```
[the remark this answers is not in the document]
Comment (keith-turner) [reply core/src/main/java/org/apache/accumulo/core/fate/FateId.java:39]: opened #4391

File: test/src/main/java/org/apache/accumulo/test/functional/AmpleConditionalWriterIT.java [hunk 3]
```diff
@@ -1086,15 +1083,15 @@ public void testCompactedAndFlushIdFilter() {
       ConditionalTabletsMutatorImpl ctmi = new ConditionalTabletsMutatorImpl(context);
       Set<TabletMetadataFilter> filter = Set.of(new TestTabletMetadataFilter());
       FateInstanceType type = FateInstanceType.fromTableId(tid);
-      FateId fateId34L = FateId.from(type, 34L);
-      FateId fateId987L = FateId.from(type, 987L);
+      FateId fateId1 = FateId.from(type, UUID.randomUUID());
```
[the remark this answers is not in the document]
Comment (kevinrr888) [reply test/src/main/java/org/apache/accumulo/test/functional/AmpleConditionalWriterIT.java:1086]: Yes, and I also double checked that the variables remained consistent

**Pull request merged**